daft/io/lance/lance_scan.py

```diff
@@ -46,11 +46,8 @@ def _lancedb_count_result_function(
         logger.debug("Using metadata for counting all rows (no filters)")
         count = ds.count_rows()
     else:
-        # TODO: If filters are provided, we need to apply them after counting
         logger.debug("Counting rows with filters applied")
-        scanner = ds.scanner(filter=filters)
-        for batch in scanner.to_batches():
-            count += batch.num_rows
+        count = ds.count_rows(filter=filters)
```

Comment:
Nit: we can just do `count = ds.count_rows(filter=filter)` no matter if the filter is None or not, and remove this if-else